build-tools/packages/build-tools/src/typeValidator/testGeneration.ts

```diff
@@ -15,30 +15,25 @@ export function buildTestCase(
 	useType: TestCaseTypeData,
 	isCompatible: boolean,
 	typePreprocessor: string,
-) {
+): string[] {
 	if (!isCompatible && (getAsType.removed || useType.removed)) {
-		return "";
+		return [];
 	}
 
-	const getSig = `get_${getAsType.prefix}_${getFullTypeName(getAsType).replace(".", "_")}`;
-	const useSig = `use_${useType.prefix}_${getFullTypeName(useType).replace(".", "_")}`;
-	const expectErrorString = "    // @ts-expect-error compatibility expected to be broken";
+	const expectErrorString = "// @ts-expect-error compatibility expected to be broken";
 	const testString: string[] = [];
 
-	testString.push(`declare function ${getSig}():`);
-	testString.push(`    ${toTypeString(getAsType.prefix, getAsType, typePreprocessor)};`);
-	testString.push(`declare function ${useSig}(`);
-	testString.push(
-		`    use: ${toTypeString(useType.prefix, useType, typePreprocessor)}): void;`,
-	);
-	testString.push(`${useSig}(`);
 	if (!isCompatible) {
 		testString.push(expectErrorString);
 	}
-	testString.push(`    ${getSig}());`);
+	testString.push(
+		`declare type ${getAsType.prefix}_as_${useType.prefix}_for_${getFullTypeName(
+			getAsType,
+		)} = requireAssignableTo<${toTypeString(
+			getAsType.prefix,
+			getAsType,
+			typePreprocessor,
+		)}, ${toTypeString(useType.prefix, useType, typePreprocessor)}>`,
+	);
 	return testString;
 }
-
-function getFullTypeName(typeData: TypeData) {
-	return `${typeData.kind}_${typeData.name}`;
-}
```

Comment:
Was this moved elsewhere in an earlier change and not cleaned up? Just double-checking.

Reply:
This code was duplicated, and since my previous change fixed the other copy to handle dots the type name, I deleted this copy of it (which I think was just in generating the doc comments since the other copy was used for generating the code). I don't know how exactly we ended up with the duplication, but both versions were getting used in different places, and I suspect it was part of previous changes which duplicated a lot of the type tests (and continued to use both copies of a lot of it)